packages/internal/bridge/sdk/src/lib/utils.ts

```diff
@@ -72,6 +72,52 @@ export function isWrappedIMX(token: FungibleToken, source: string) {
   return token.toUpperCase() === getWrappedIMX(source).toUpperCase();
 }
 
+/**
+ * Assumes the destination chain is correct (i.e. it doesn't check the destination chain)
+ */
+export function isDeposit(sourceChainId: string, bridgeInstance: BridgeInstance): boolean {
+  return sourceChainId === bridgeInstance.rootChainID;
+}
+
+/**
+ * Assumes the destination chain is correct (i.e. it doesn't check the destination chain)
+ */
+export function isWithdraw(sourceChainId: string, bridgeInstance: BridgeInstance): boolean {
+  return sourceChainId === bridgeInstance.childChainID;
+}
+
+export function isWithdrawNotWrappedIMX(
+  token: FungibleToken,
+  sourceChainId: string,
+  bridgeInstance: BridgeInstance,
+): boolean {
+  return isWithdraw(sourceChainId, bridgeInstance) && !isRootIMX(token, sourceChainId);
+}
+
+export function isWithdrawWrappedIMX(
+  token: FungibleToken,
+  sourceChainId: string,
+  bridgeInstance: BridgeInstance,
+): boolean {
+  return isWithdraw(sourceChainId, bridgeInstance) && isRootIMX(token, sourceChainId);
+}
+
+/**
+ * This is the same as `isDeposit`, but takes an extra parameter - the `destinationChainId` - to check that both
+ * of the chain IDs are correct.
+ */
+export function isValidDeposit(sourceChainId: string, destinationChainId: string, bridgeInstance: BridgeInstance) {
+  return sourceChainId === bridgeInstance.rootChainID && destinationChainId === bridgeInstance.childChainID;
+}
+
+/**
+ * This is the same as `isWithdraw`, but takes an extra parameter - the `destinationChainId` - to check that both
+ * of the chain IDs are correct.
+ */
+export function isValidWithdraw(sourceChainId: string, destinationChainId: string, bridgeInstance: BridgeInstance) {
+  return sourceChainId === bridgeInstance.childChainID && destinationChainId === bridgeInstance.rootChainID;
+}
```

Comment:
is there a reason we wouldn't replace `isDeposit()` and `isWithdrawal()` with these? The additional verification of the destination chain is beneficial for integrity as well.

Reply:
Yeah I tend to agree. I made the current changes because I was compartmentalising the code that currently existed in `tokenBridge.ts`. But thinking about it more, we should probably not do that here.